leaflet-treering.js

```diff
@@ -1159,6 +1163,236 @@ function AnnotationAsset(Lt) {
   };
 }
 
+/**
+ * Scale bar for orientation & screenshots
+ * @constructor
+ * @param {LTreering} - Lt
+ */
+function ScaleBarCanvas (Lt) {
+  var scaleBarDiv = document.createElement('div');
+  var nativeWindowWidth = document.getElementById("map_container").clientWidth;
+  scaleBarDiv.innerHTML =
+      '<div id="scale-bar-div"> \
+       <canvas id="scale-bar-canvas" width="' + nativeWindowWidth + '" height="100"></canvas> \
+       </div>';
+  document.getElementsByClassName('leaflet-bottom leaflet-left')[0].appendChild(scaleBarDiv);
+
+  var canvas = document.getElementById("scale-bar-canvas");
+  var ctx = canvas.getContext("2d");
+
+  var map = Lt.viewer;
+
+  ScaleBarCanvas.prototype.load = function () {
+    var pixelWidth;
+    map.eachLayer(function (layer) {
+      if (layer.options.maxNativeZoom) {
+        var leftMostPt = layer.options.bounds._southWest;
+        var rightMostPt = layer.options.bounds._northEast;
+        pixelWidth = map.project(rightMostPt, Lt.getMaxNativeZoom()).x;
+      }
+    });
+
+    var windowZoom = true; // used to set initial zoom level
+    function modifyScaleBar() {
+      ctx.clearRect(0, 0, nativeWindowWidth, 100);
+
+      if (windowZoom) {
+        this.initialZoomLevel = map.getZoom();
+        windowZoom = false;
+      }
+
+      var metricWidth = pixelWidth / Lt.meta.ppm;
+      var currentZoomLevel = map.getZoom();
+      var zoomExponentialChange = Math.pow(Math.E, -0.693 * (currentZoomLevel - this.initialZoomLevel)); // -0.693 found from plotting zoom level with respect to length in excel then fitting expoential eq.
```

Comment:
This is really clever, but I also worry that it might be assuming a level of consistency that doesn't actually exist. Will ask about testing in the main PR

Reply:
When tested on chrome/firefox with a mouse & laptop touch screen & also on safari with an ipad, it seems to be correct give  or take a few millimeters. Its more accurate the more zoomed in you are.

Reply:
cool. Do we need to inform users that it's intended as a guide and not an exact cacluation?

Reply:
Talking w/ Dan, we think adding '~' for approximately will suffice enough. Kinda a cheaty work around with all the pixel differences.